modmail/extensions/utils/error_handler.py

```diff
@@ -0,0 +1,195 @@
+import logging
+import re
+import typing
+
+import discord
+import discord.errors
+from discord.ext import commands
+
+from modmail.bot import ModmailBot
+from modmail.log import ModmailLogger
+from modmail.utils.cogs import BotModes, ExtMetadata, ModmailCog
+from modmail.utils.extensions import BOT_MODE
+
+
+logger: ModmailLogger = logging.getLogger(__name__)
+
+EXT_METADATA = ExtMetadata()
+
+ERROR_COLOUR = discord.Colour.red()
+
+ERROR_TITLE_REGEX = re.compile(r"(?<=[a-zA-Z])([A-Z])(?=[a-z])")
+
+ANY_DEV_MODE = BOT_MODE & (BotModes.DEVELOP.value + BotModes.PLUGIN_DEV.value)
+
+
+class ErrorHandler(ModmailCog, name="Error Handler"):
+    """Handles all errors across the bot."""
+
+    def __init__(self, bot: ModmailBot):
+        self.bot = bot
+
+    @staticmethod
+    def error_embed(message: str, title: str = None) -> discord.Embed:
+        """Create an error embed with an error colour and reason and return it."""
+        return discord.Embed(message, colour=ERROR_COLOUR, title=title or "Error Occurred")
```

Comment:
`title` should not be nullable here. Let's enforce some kind of automated naming, like derived from the exception name. Plus, it's not typehinted as Optional.

Reply:
There is a place where its already dervied from the exception class. That exists elsewhere, and is used for most names. I suppose I could remove the default here, since its only used in one place and call the method with that default. 

However, if making it required, I'm switching the argument order as well.